### PR TITLE
[SPARK-6469] Improving documentation on YARN local directories usage

### PR DESCRIPTION
**Pull request by preaudc**
Clarify the local directories usage in YARN

### CODE REVIEW

**Comment by preaudc** [comment]
I don't know if this is normal that all commits show up in the PR, I'm not yet familiar with this tool... Anyway, only the last one is relevant for this PR, and that's what can be seen in the "Files changed" tab.

File: docs/running-on-yarn.md [hunk 1]
```diff
@@ -274,6 +274,6 @@ If you need a reference to the proper location to put log files in the YARN so t
 # Important notes
 
 - Whether core requests are honored in scheduling decisions depends on which scheduler is in use and how it is configured.
-- The local directories used by Spark executors will be the local directories configured for YARN (Hadoop YARN config `yarn.nodemanager.local-dirs`). If the user specifies `spark.local.dir`, it will be ignored.
+- In yarn-cluster mode, the local directories used by the Spark executors and the Spark driver will be the local directories configured for YARN (Hadoop YARN config `yarn.nodemanager.local-dirs`). If the user specifies `spark.local.dir`, it will be ignored. In yarn-client mode, the Spark executors will use the local directories configured for YARN while the Spark driver will use those defined in `spark.local.dir` (this is because the Spark driver does not run on the YARN cluster in yarn-client mode, only the Spark executors do).
```
Comment (srowen) [inline docs/running-on-yarn.md]: Nits: you can back-tick-quote code-like values like `yarn-cluster`. The last parenthetical can just be a sentence. Yeah it's normal to show all the commits you made to your branch. They will be squashed on merge. You can also squash and force-push your branch if desired, though it's not that important.

**Comment by tgravescs** [comment]
Thanks for providing the pull request. The text here looks good.  I'm +1.  I'll let @srowen finish his review.

**Comment by sryza** [comment]
Mind changing the title to reflect that this is a documentation fix?

**Comment by preaudc** [comment]
Sure, is the new title ok?

**Comment by sryza** [comment]
New title looks great to me

**Comment by andrewor14** [comment]
LGTM merging this into master thanks.

**Pull request closed without merging**